File: benchmark/src/main/java/de/hhu/bsinfo/observatory/benchmark/CleanupPhase.java
package de.hhu.bsinfo.observatory.benchmark;

import de.hhu.bsinfo.observatory.benchmark.result.Status;

public class CleanupPhase extends SetupPhase {

    CleanupPhase(Benchmark benchmark) {
        super(benchmark);
    }

    @Override
    protected Status execute() {
        return getBenchmark().cleanup();
    }
}

File: cpp/src/observatory/phase/CleanupPhase.cpp
#include "CleanupPhase.h"
#include <observatory/Benchmark.h>

#include <utility>

namespace Observatory {

CleanupPhase::CleanupPhase(Benchmark &benchmark) :
        BenchmarkPhase(benchmark) {}

const char* CleanupPhase::getName() {
    return "CleanupPhase";
}

Status CleanupPhase::execute() {
    return getBenchmark().cleanup();
}

}